spf.go

```diff
@@ -1,20 +1,52 @@
 package spf
 
 import (
-	"fmt"
+	"errors"
 	"net"
 	"strconv"
 	"strings"
+)
 
-	"github.com/miekg/dns"
+// Errors could be used for root couse analysis
+var (
+	ErrDNSTemperror      = errors.New("temporary DNS error")
+	ErrDNSPermerror      = errors.New("permanent DNS error")
+	ErrInvalidDomain     = errors.New("invalid domain name")
+	errInvalidCIDRLength = errors.New("invalid CIDR length")
+	errDNSLimitExceeded  = errors.New("limit exceeded")
 )
```

Comment:
What do you mean by saying "recovery logic implementation"? 
Production or not, when something bad happens I want to at least know for which particular case the library returned an error. With predefined errors I feel we are using some information of what was the input, or which part of the workflow failed (there may be lots of DNS calls and one of them down there the stack may be failing or something bad may be happening).